clang/lib/CIR/CodeGen/CIRGenBuiltin.cpp

```diff
@@ -100,6 +100,33 @@ RValue CIRGenFunction::emitBuiltinExpr(const GlobalDecl &gd, unsigned builtinID,
     mlir::Value complex = builder.createComplexCreate(loc, real, imag);
     return RValue::get(complex);
   }
+
+  case Builtin::BI__builtin_expect:
+  case Builtin::BI__builtin_expect_with_probability: {
+    mlir::Value argValue = emitScalarExpr(e->getArg(0));
+    mlir::Value expectedValue = emitScalarExpr(e->getArg(1));
+
+    mlir::FloatAttr probAttr;
+    if (builtinIDIfNoAsmLabel == Builtin::BI__builtin_expect_with_probability) {
+      llvm::APFloat probability(0.0);
+      const Expr *probArg = e->getArg(2);
+      bool evalSucceeded =
```

Comment:
```suggestion
      [[maybe_unused]] bool evalSucceeded =
```
There is a proposal under review (https://github.com/llvm/llvm-project/pull/142850) to make this the standard way of handling variables that are used only in asserts. More discussion here (https://discourse.llvm.org/t/coding-standard-deprecate-use-of-void-cast-to-suppress-unused-variable-warnings/86705), but I expect this to go through, so we should start following this practice.